test/e2e/trafficcontrol_test.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"antrea.io/antrea/pkg/apis/crd/v1alpha2"
+	agentconfig "antrea.io/antrea/pkg/config/agent"
+)
+
+var (
+	fakeExternalPod   = "fake-pod"
+	fakeExternalPodNS = "fake-ns"
+	fakeExternalIP    = "1.1.1.1"
+	fakeExternalGW    = "1.1.1.254"
+
+	vni           = int32(1)
+	dstVXLANPort  = int32(4790)
+	dstGENEVEPort = int32(6082)
+	greKey        = int32(2222)
+
+	labels = map[string]string{"tc-e2e": "agnhost"}
+
+	testNode      string
+	antreaPodName string
+)
+
+func TestTrafficControl(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+	skipIfNotIPv4Cluster(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	ac := func(config *agentconfig.AgentConfig) {
+		config.FeatureGates["TrafficControl"] = true
+	}
+
+	if err := data.mutateAntreaConfigMap(nil, ac, true, true); err != nil {
+		t.Fatalf("Failed to enable TrafficControl feature: %v", err)
+	}
+
+	testNode = controlPlaneNodeName()
+	antreaPodName, err = data.getAntreaPodOnNode(testNode)
+	if err != nil {
+		t.Fatalf("Error when retrieving the name of the Antrea Pod running on Node '%s': %v", testNode, err)
+	}
+
+	// Create a host network Pod to fake external network.
```

Comment:
why it needs external network to test this feature?

Reply:
I thought of we need another namespace to fake another network to receive the tunnelled packets.  I'm not sure if we can just use the root network namespace of the Node to received the tunnelled packets. I'll make a test.

Reply:
I think any network endpoint can receive the traffic

Reply:
I found that we can create a VXLAN tunnel on the OVS running in a Node and create another VXLAN tunnel, and they uses the same port. Of course, we can use another Node or Pod to receive the traffic, but not the local Node.

Reply:
you don't need to create VXLAN tunnel using OVS. Just use `ip link` to create a vxlan and use a different port.

Reply:
> you don't need to create VXLAN tunnel using OVS. Just use ip link to create a vxlan and use a different port.

You mean the tunnel peer to receive the mirrored packets? If so, I have tried this. Using `ip link` to add the a vxlan tunnel with different port cannot receive the mirrored packets.